### PR TITLE
integrate Search API to display restaurant options

### PR DESCRIPTION
**Pull request by graceehuu**
integrated Search API
displays restaurants within 10,000 m of center
fixed: when user clicks on a marker, displays the clicked marker name in panel

image

### CODE REVIEW

File: capstone/src/main/webapp/script.js [hunk 1]
```diff
@@ -12,11 +12,19 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+window.addEventListener('DOMContentLoaded', (event) => {
+  if (document.getElementById('map')) {
+    createMap();
+  }
+});
+
+var map;
```
Comment (gannerr) [inline capstone/src/main/webapp/script.js:21]: could we put these variables inside the function?

Comment (graceehuu) [reply capstone/src/main/webapp/script.js:21]: these vars are used in other functions, but I'll reduce the scope of the vars

Comment (gannerr) [reply capstone/src/main/webapp/script.js:21]: oh i misread, mb

File: capstone/src/main/webapp/script.js [hunk 2]
```diff
@@ -97,61 +106,42 @@ function createMap() {
...
-        zIndex: restaurant.zIndex
-    });
-    marker.addListener('click', toggleBounce);
-    marker.addListener('click', displayPanel(restaurant.name));
-    
-    marker.setMap(map);
+function callback(results, status) {
+  if (status == google.maps.places.PlacesServiceStatus.OK) {
+    for (var i = 0; i < results.length; i++) {
+      // TODO: change condition to check if matches filter
```
Comment (terencedignon) [inline capstone/src/main/webapp/script.js:127]: Can we add a Github issue to the todo? For instance TODO(issuenum):  update condition to check if matches filter

File: capstone/src/main/webapp/script.js [hunk 3]
```diff
@@ -12,11 +12,20 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+window.addEventListener('DOMContentLoaded', (event) => {
+  if (document.getElementById('map')) {
+    createMap();
+  }
+});
+
+var map;
```
Comment (terencedignon) [inline capstone/src/main/webapp/script.js:21]: I'm wondering if there's a way we can encapsulate this, and then export the functions. If it's too much trouble, I think it's best practice to indicate that these are private variables by appending a _ to them.

**Comment by terencedignon** [comment]
Hi Grace, is this PR attached to an issue? If not, could we make an issue to address what is happening here and then tag it in the PR description with "#closes {issuenum}"

**Comment by casanted** [comment]
looks good to me!

File: capstone/src/main/webapp/script.js [hunk 4]
```diff
@@ -97,61 +106,42 @@ function createMap() {
...
-        position: {lat: restaurant.latitude, lng: restaurant.longitude},
-        zIndex: restaurant.zIndex
-    });
-    marker.addListener('click', toggleBounce);
-    marker.addListener('click', displayPanel(restaurant.name));
-    
-    marker.setMap(map);
+function callback(results, status) {
+  if (status == google.maps.places.PlacesServiceStatus.OK) {
+    for (var i = 0; i < results.length; i++) {
```
Comment (terencedignon) [inline capstone/src/main/webapp/script.js:126]: let :)

**Pull request merged**